### DISCUSSION TITLE
对于 ethStorage 的技术实现是怎么样的？

### DISCUSSION THREAD

**Discussion by jjeejj** [IntensiveCoLearning/Web3-URL#134] [Q&A]
ethStorage 是在链上保存 blob 文件，它的技术实现什么样的？ 有文件大小的限制的

**Comment by qizhou** [comment]
> ethStorage 是在链上保存 blob 文件，它的技术实现什么样的？ 有文件大小的限制的

EthStorage是链下保存BLOB数据，链上合约保存BLOB哈希和链下数据保存的验证（通过ZK存储证明）。每个BLOB是按照以太坊标准EIP-4844，大概是128KB。但是可以多个BLOB来组成一个文件，所以理论上是没有文件大小限制的。

**Reply by jjeejj** [reply]
链下保存BLOB数据，代表还是中心化保存吗？ 链下提供存储的 服务商跑路了，数据就没有了

**Reply by zedz** [reply]
似乎好像看ethStorage.io，链下提供存储的服务商应该是很多个（大量的副本），除非所有的服务商都跑路？

**Reply by qzhodl** [reply]
> 似乎好像看ethStorage.io，链下提供存储的服务商应该是很多个（大量的副本），除非所有的服务商都跑路？

是的，EthStorage会激励多个存储服务商备份多个副本。
